pkg/github/commenter/ghcommenter.go

```diff
@@ -213,25 +213,83 @@ func (ghc *GitHubCommenter) UpdatePendingCommentRecords(org, repo string, number
 	}
 }
 
-func (ghc *GitHubCommenter) ClearPendingRecord(org, repo string, number int, sha string, record *models.PullRequestComment) {
+func (ghc *GitHubCommenter) ValidatePendingRecordComment(org, repo string, number int, sha string, commentType models.CommentType) (bool, error) {
+	// check for this record in our table
+	// check to see what the last time we attempted to write a comment was
+	// if within threshold then skip
+	// if not then update the last attempt time
+	// if we don't have the record then skip
+	pullRequestComment, err := ghc.getPendingRecord(org, repo, number, sha, commentType)
+
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return false, nil
+		}
+
+		return false, err
+	}
+
+	// default is wait 10 minutes but then increase the wait by the number of errors we have encountered for this record
+	waitMillis := int64(600000 * (pullRequestComment.FailedCommentAttempts + 1))
+
+	if (time.Now().UnixMilli() - pullRequestComment.LastCommentAttempt.UnixMilli()) < waitMillis {
+		return false, nil
+	}
+
+	res := ghc.dbc.DB.Model(&models.PullRequestComment{}).Where("org = ? AND repo = ? AND pull_number = ? AND sha = ? AND comment_type = ?  ", pullRequestComment.Org, pullRequestComment.Repo, pullRequestComment.PullNumber, pullRequestComment.SHA, pullRequestComment.CommentType).Update("last_comment_attempt", time.Now())
+	if res.Error != nil {
+		return false, res.Error
+	}
+
+	return true, nil
+}
+
+func (ghc *GitHubCommenter) UpdatePendingRecordErrorCount(org, repo string, number int, sha string, commentType models.CommentType) error {
+	pullRequestComment, err := ghc.getPendingRecord(org, repo, number, sha, commentType)
+
+	if err != nil {
+		return err
+	}
+
+	// this would be the tenth failure so just delete it
+	if pullRequestComment.FailedCommentAttempts > 8 {
```

Comment:
I don't understand this.  You say tenth, yet you are mentioning `8`

Reply:
The incoming failure is the 10th.  It could be written as >= 9 but I went with > 8 (meaning it is likely 9) and the incoming attempt is the 10th so bail.